### ISSUE TITLE
Landing page not prepending /app to sub application routing

### ISSUE THREAD

**Issue by mairead** [epimorphics/hmlr-linked-data#52]
The routing on http://hmlr-dev-pres.epimorphics.net/ to UKHPI, PPD and standard reports is missing the /app prepended to the URL so the links to each application are returning a 404 not found

Notes from Mihajlo:
Looking at the ppd explored dockerfile I see
`ENV RELATIVE_URL_ROOT="/app/ppd"`
Whereas in landing you are setting
`ENV RELATIVE_URL_ROOT="/"`
You can try setting that to /app and see whether the navigation links work. If that is the case then we can route the landing page traffic to the landing app at landing:3000/app and that should fix it.

**Comment by ijdickinson** [comment]
> Whereas in landing you are setting
> ENV RELATIVE_URL_ROOT="/"

That's correct. The relative URL root is the path at which the application itself appears, relative to the domain. PPD appears as `https://landegistry.data.gov.uk/app/ppd`, whereas the landing page appears as `https://landregistry.data.gov.uk/`

**Comment by ijdickinson** [comment]
Actually, I'm wrong. The landing app is actually published as `/app/root`, so the `RELATIVE_URL_ROOT` env var for production should be `/app/root`. I believe in the current apps we use routing in the front-end web server to direct traffic for `/` to `/app/root`.

**Comment by joescottdave** [comment]
Does the below adequately summarise the expectation of the proxy server/the setup of these applications?

=======

Each of the four applications will run in production mode by setting - `RAILS_ENV="production"` - in a container with port 3000 exposed and with Rails set to serve static assets - using `RAILS_SERVE_STATIC_ASSETS="true"` - from the application's respective `RELATIVE_URL_ROOT` shown in the table below.  The proxy should also send requests for the root ('/') to lr-landing.

| Application | Relative URL root | Other routes |
| ------------ | ------------------- | -------------- |
| lr-landing | '/app/root' | '/' |
| ppd-explorer | '/app/ppd' | N/A |
| standard-reports | '/app/standard-reports' | N/A |
| ukhpi | '/app/ukhpi' | N/A |
| qonsole | '/app/qonsole' | N/A |

**Comment by ijdickinson** [comment]
Looks good. I'd say "The proxy should ~also send~ alias requests for the root ...".

**Comment by joescottdave** [comment]
I think this maybe wrong for qonsole? That route is controlled by lr-landing?

**Comment by ijdickinson** [comment]
Qonsole is controlled by lr-landing, yes, but I wasn't sure if there were explicit routes for it in the FE web server. I guess it depends what the default route is. Iow where does: `/some/random/path` go to? If everything *except* PPD, UKHPI or standard reports gets sent to lr-landing, then there wouldn't need to be a separate row in the table for `/app/qonsole`

**Comment by joescottdave** [comment]
Okay so after talking to @mika018 and getting him to alias requests for qonsole, I think the above table should probably look more like this. @andrew-pickin-epi ?

| Outside request | served by application | at route |
| ---- | ---- | ---- |
| '/' | lr-landing | '/' |
| '/app/root' | lr-landing | '/' |
| '/app/qonsole' | lr-landing | '/qonsole' |
| '/app/ppd' | ppd-explorer | '/' |
| '/app/standard-reports' | standard-reports | '/' |
| '/app/ukhpi' | ukhpi | '/' |

**Comment by andrew-pickin-epi** [comment]
I don't think this table answers @ijdickinson's question re `/some/random/path`. How does this match the first line but `/app/ppd` doesn't? In this table is Does request '/' mean everything starting with '/' obviously not.  So you have to stop once there is a match and have some precedence and stop when you have a match.

It's also a pain if we have to split an application's configuration.  So `lr-landing` needs to be all together. Thus:

| Outside request | served by application | at route |
| ---- | ---- | ---- |
| `^/app/ppd/` | ppd-explorer | `/` |
| `^/app/standard-reports/` | standard-reports | `/` |
| `^/app/ukhpi/` | ukhpi | `/` |
| `^/app/qonsole/` | lr-landing | `/qonsole` |
| `^/app/root/` | lr-landing | `/` |
| `^/` | lr-landing | `/` |

I'm not convinced about the last two entries here. Is the `lr-landing` really agnostic to `/` and `/app/root` requests or just strips away leading `/app/root`?

**Comment by joescottdave** [comment]
The current production setup routes '/' and '/app/root' to the same place (lr-landing index) without rewriting the original request. I assume we want to preserve that behaviour.

**Comment by joescottdave** [comment]
Going to continue as part of a new issue #79
